src/DIRAC/WorkloadManagementSystem/Optimizer/InputData.py

```diff
@@ -0,0 +1,431 @@
+from DIRAC.WorkloadManagementSystem.Client import JobMinorStatus
+from DIRAC.WorkloadManagementSystem.Client.JobState.JobState import JobState
+from DIRAC.WorkloadManagementSystem.OptimizerAdministrator.Optimizer import Optimizer
+
+
+class InputData(Optimizer):
+    """
+    The specific Optimizer must provide the following methods:
+      - optimize() - the main method called for each job
+    """
+
+    def __init__(self):
+        """Initialize specific parameters for InputData executor."""
+
+        # TODO: the three code lines bellow should be modified to eliminate the ex_getOption method from the executor framework
+        # self.failedMinorStatus = self.ex_getOption("/FailedJobStatus", "Input Data Not Available")
+        # # this will ignore failover SE files
+        # self.checkFileMetadata = self.ex_getOption("CheckFileMetadata", True)
+        # # flag to require Input Data lookup with a user proxy
+        # self.checkWithUserProxy = self.ex_getOption("CheckWithUserProxy", False)
+
+        self.__dataManDict = {}
+        self.__fcDict = {}
+        self.__SEToSiteMap = {}
+        self.__lastCacheUpdate = 0
+        self.__cacheLifeTime = 600
+
+        # Note: this is a default, that right now is generically the default for user jobs, at least for main DIRAC users
+        # (since this now doesn't run for production jobs)
+        # But this should probably be replaced by what the job actually request.
+        # The problem is that the InputDataPolicy is not easy to get (a JDL parameter).
+        # This may be used but clear how now
+        # cls.__connectionLevel = 'PROTOCOL'
+
+    def __getDataManager(self, vo):
+        if vo in self.__dataManDict:
+            return self.__dataManDict[vo]
+
+        try:
+            self.__dataManDict[vo] = DataManager(vo=vo)
+        except Exception:
+            msg = "Failed to create DataManager"
+            self.log.exception(msg)
+            return None
+        return self.__dataManDict[vo]
+
+    def __getFileCatalog(self, vo):
+        if vo in self.__fcDict:
+            return self.__fcDict[vo]
+
+        try:
+            self.__fcDict[vo] = FileCatalog(vo=vo)
+        except Exception:
+            msg = "Failed to create FileCatalog"
+            self.log.exception(msg)
+            return None
+        return self.__fcDict[vo]
+
+    def optimize(self, jobState: JobState):
+        """This is the method that needs to be implemented by each and every Executor
+
+        This optimizer will run if and only if it is needed:
+          - it will run only if there are input files
+          - for production jobs this can be skipped,
+            since the logic is already applied by the transformation system, via the TaskManagerPlugins
+        """
+        # Is it a production job?
+        result = jobState.getAttribute("JobType")
+        if not result["OK"]:
+            self.log.error("Could not retrieve job type", result["Message"])
+            return result
```

Comment:
It can, but the issue is that if a production job manifest is misconfigured and contains the InputData optimizer, the executor will be executed and it will return S_ERROR instead of S_OK in the actual code.